### PR TITLE
Support sync and async gunicorn workers

### PR DESCRIPTION
**Pull request by pascalknapen**
When running airflow v1.5.2 on AWS with Elastic Load Balancer (ELb), you experience extreme high latency due to the interplay between gunicorn and  ELB as described here: https://forums.aws.amazon.com/thread.jspa?messageID=419138. To resolve this I added support to airflow to support both sync and async workers.

### CODE REVIEW

**Comment by artwr** [comment]
:+1:  For better PR than mine with the extras.

File: airflow/configuration.py [hunk 1]
```diff
@@ -157,8 +158,12 @@ class AirflowConfigException(Exception):
 # Secret key used to run your flask app
 secret_key = temporary_key
 
-# number of threads to run the Gunicorn web server
-thread = 4
+# Number of threads to run the Gunicorn web server
+threads = 4
```
Comment (artwr) [inline airflow/configuration.py]: Might be worth renaming to workers.

File: airflow/bin/cli.py [hunk 2]
```diff
@@ -600,6 +600,10 @@ def get_parser():
         type=int,
         help="Number of threads to run the webserver on")
     parser_webserver.add_argument(
+        "-k", "--workerclass",
+        default=configuration.get('webserver', 'WORKER_CLASS'),
```
Comment (artwr) [inline airflow/bin/cli.py:604]: `choices=['sync', 'eventlet', 'gevent', 'tornado'],` We can constrain this since those are the only options.

**Comment by pascalknapen** [comment]
Made the following recommended changes: 
- adding argparse choices for worker_class
- renamed 'threads' to 'workers'

**Comment by artwr** [comment]
LGTM

**Comment by mtustin-handy** [comment]
:heart:

**Pull request merged**